src/pseudopeople/utilities.py

```diff
@@ -192,6 +192,35 @@ def cleanse_integer_columns(column: pd.Series) -> pd.Series:
     return column
 
 
+def number_of_tokens_per_string(s1, s2):
+    """
+    Calculates the number of tokens in each string of a series.
+    s1 is a pd.Series of tokens and we want to see how many tokens exist in each
+    string of s2.
+    """
+
+    s2 = s2.astype(str)
+    strings = s2.unique()
+    tokens_per_string = pd.Series(
+        (sum(occurrences(s, str(token)) for token in s1) for s in strings), index=strings
+    )
+
+    number_of_tokens = s2.map(tokens_per_string)
+    number_of_tokens.index = s2
+    return number_of_tokens
+
+
+# https://stackoverflow.com/a/2970542/
+def occurrences(string, sub):
+    count = start = 0
+    while True:
+        start = string.find(sub, start) + 1
```

Comment:
Does this properly account for case, e.g. 'a' vs 'A'?

Reply:
No - which is actually correct. As of now none of our noise functions are case insensitive. All of our data for tokens that can be noised are case sensitive so 'Aaa' and 'a' would only have a count of 2 vs is 3. This is something I raised with RT and we are going to discuss further later.